zerver/models.py

```diff
@@ -3831,7 +3831,7 @@ class CustomProfileField(models.Model):
         (SELECT, gettext_lazy("List of options"), validate_select_field, str, "SELECT"),
     ]
     USER_FIELD_TYPE_DATA: List[UserFieldElement] = [
-        (USER, gettext_lazy("Person picker"), check_valid_user_ids, ast.literal_eval, "USER"),
+        (USER, gettext_lazy("Person picker"), check_valid_user_ids, orjson.loads, "USER"),
```

Comment:
@andersk Does this look good? 47913fb091794830c54eac2927e26bce92cfced1 changed this from `eval` to `ast.literal_eval`, so I'm wondering if that was due to uncertainty about what strings exactly this converter deals with or if there is reason `orjson.loads` shouldn't be used?

Reply:
My explanation to Tim at the time was:

> Part of what makes the `eval` confusing is that every deserialization should have an equal and opposite serialization—so where is it and why does it only affect list fields without adding extra quotes to string fields? I think the answer is that this line in `do_update_user_custom_profile_data_if_changed` is not type-correct; it’s assigning `Union[int, str, List[int]]` to a `TextField`, which I assume implicitly stringifies with `__str__`.
>
> ```python
>             field_value.value = field['value']
> ```
>
> …
>
> So we can probably just replace the `eval` with `ast.literal_eval` or `json.loads`?
>
> I think if we did `json.loads`, we should also do the restructuring so that there’s a corresponding `json.dumps` rather than an implicit `__str__`.
>
> Whereas `ast.literal_eval` is an admission that the rest of the code is weird.

The same reasoning still applies to your PR. `orjson.loads()` is not the semantic opposite of `str()`—it just accidentally works for the specific case of lists of integers. If we want to use `orjson.loads()`, we should put an `orjson.dumps()` on the other side.